### ISSUE TITLE
Enum type aliasing an extension within VkPipelineStageFlagBits2

### ISSUE THREAD

**Issue by spnda** [KhronosGroup/Vulkan-Docs#1814]
Within vk.xml, and by extension the Vulkan-Headers, one can find that the enum value `VK_PIPELINE_STAGE_2_TRANSFER_BIT` actually aliases a value from an extension: `VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR`. As the enum from the extension then again just aliases a value from Vulkan core, I would deem this a typo and unintended.
```xml
<enum bitpos="12" name="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT"/>
<enum name="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR" alias="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT"/>
<enum name="VK_PIPELINE_STAGE_2_TRANSFER_BIT" alias="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR"/>
```

In my opinion, `VK_PIPELINE_STAGE_2_TRANSFER_BIT` should directly alias `VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT`, instead of going through a value from another extension (Though I ask myself why anyone differentiates between the two anyway, they seem the same?). Within the whole enum, and all other enums, this does not seem to occur again, further suggesting this is a typo. I came across this writing a Vulkan header generator that assumes that any value from an extension comes *after* the core values.

**Comment by krOoze** [comment]
`VkPipelineStageFlagBits2KHR` verbatim copies all bits from `VkPipelineStageFlagBits`:

```
<enum bitpos="12" name="VK_PIPELINE_STAGE_2_TRANSFER_BIT_KHR"/>
```

But the extension also adds things like `VK_PIPELINE_STAGE_2_COPY_BIT_KHR`, so the name is misleading. So newer alias `VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR` is added:

```
<enum bitpos="12" name="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR"/>
<enum name="VK_PIPELINE_STAGE_2_TRANSFER_BIT_KHR"" alias="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR"/>
```

Extension gets promoted to 1.3. So aliases to everything without `_KHR` suffix are added:

```
<enum bitpos="12" name="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT"/>
<enum name="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR"" alias="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT"/>

<enum name="VK_PIPELINE_STAGE_2_TRANSFER_BIT"" alias="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT_KHR"/>
<enum name="VK_PIPELINE_STAGE_2_TRANSFER_BIT_KHR"" alias="VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT"/>
```

> should directly alias `VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT`

Now, they **could** all be aliased to `VK_PIPELINE_STAGE_2_ALL_TRANSFER_BIT`. But the xml schema does not give the guarantee that the alias points to real value and not another alias, does it?

**Comment by oddhack** [comment]
It's a fair point. Just an informal convention not required by the schema, but we do follow it in other cases of multiple promotion. Will try to correct this soon.

> (Though I ask myself why anyone differentiates between the two anyway, they seem the same?)

I don't remember specifically in this case but it is usually a case where we didn't follow our own naming conventions and fixed it later.

**Comment by oddhack** [comment]
This should be fixed in the 1.3.211 spec update.
